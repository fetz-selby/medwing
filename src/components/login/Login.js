import React from 'react';
import PropTypes from 'prop-types';
import './login.css';
import icon from '../../assets/icons/user.png';

const LoginWidget = (props)=>
    <div className='login' onClick={()=>props.onLoginClicked(props.id)}>
        <div className='login-widget-container'>
            <div className='avatar-container'>
                {/* <img src={props.avatar} alt={props.name}/> */}
                <img className='avatar' src={icon} alt={props.name}/>
            </div>
            <div className='info-container'>
                <div className='info'>{props.name}</div>
            </div>
            
        </div>
        <div className='clearfix'></div>
    </div>

const Login = (props) =>
    <div className='login-container'>
        {props.users.map((user)=>
            <LoginWidget key={user.id} 
                        id={user.id} 
                        name={user.name} 
                        avatar={user.avatar} 
                        onLoginClicked={props.onLoginClicked}/>
            )}
    </div>


Login.propTypes = {
    showHeader: PropTypes.bool,
    onLoginClicked: PropTypes.func,
    users: PropTypes.array
}

export default Login;