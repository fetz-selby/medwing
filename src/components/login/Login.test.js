import React from 'react';
import {shallow} from 'enzyme';
import Login from './';

describe('Login Component', ()=>{
    it('should render users correctly', ()=>{
        const users = [{id: 1, name: 'John Doe'}, {id: 2, name : 'Jane Doe'}];

        const login = shallow(<Login users={users}/>);
        expect(login).toMatchSnapshot();
    })
})