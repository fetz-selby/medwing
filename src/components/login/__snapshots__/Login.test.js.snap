// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render users correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    users={
      Array [
        Object {
          "id": 1,
          "name": "John Doe",
        },
        Object {
          "id": 2,
          "name": "Jane Doe",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <LoginWidget
          id={1}
          name="John Doe"
        />,
        <LoginWidget
          id={2}
          name="Jane Doe"
        />,
      ],
      "className": "login-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "avatar": undefined,
          "id": 1,
          "name": "John Doe",
          "onLoginClicked": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "avatar": undefined,
          "id": 2,
          "name": "Jane Doe",
          "onLoginClicked": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <LoginWidget
            id={1}
            name="John Doe"
          />,
          <LoginWidget
            id={2}
            name="Jane Doe"
          />,
        ],
        "className": "login-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "avatar": undefined,
            "id": 1,
            "name": "John Doe",
            "onLoginClicked": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "avatar": undefined,
            "id": 2,
            "name": "Jane Doe",
            "onLoginClicked": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
